ehrapy/tools/feature_ranking/_rank_features_groups.py

```diff
@@ -320,8 +325,54 @@ def rank_features_groups(
          >>> ep.tl.rank_features_groups(adata, "service_unit")
          >>> ep.pl.rank_features_groups(adata)
     """
+    # if rank_obs_columns is indicated, layer must be None
+    if layer is not None and rank_obs_columns is not None:
+        raise ValueError("Only one of layer and rank_obs_columns can be specified.")
+
     adata = adata.copy() if copy else adata
 
+    if rank_obs_columns is not None:
+        # keep reference to original adata, needed if copy=False
+        adata_orig = adata
+        # copy adata to work on
+        adata = adata.copy()
+
+        if isinstance(rank_obs_columns, str):
+            if rank_obs_columns == "all":
+                rank_obs_columns = adata.obs.keys().to_list()
+            else:
+                raise ValueError(
+                    f"rank_obs_columns should be 'all' or Iterable of column names, not {rank_obs_columns}."
+                )
+
+        # consider adata where all columns from obs become the features, and the other features are dropped
+        if not all(elem in adata.obs.columns.values for elem in rank_obs_columns):
+            raise ValueError(
+                f"Columns `{[col for col in rank_obs_columns if col not in adata.obs.columns.values]}` are not in obs."
+            )
+
+        # if groupby in rank_obs_columns:
```

Comment:
Can we remove the outcommented dead code, please?

Reply:
Yep done